haystack/components/eval/statistical_evaluator.py

```diff
@@ -0,0 +1,131 @@
+import collections
+from enum import Enum
+from typing import Any, Dict, List, Optional
+
+from numpy import array as np_array
+from numpy import mean as np_mean
+
+from haystack import default_from_dict, default_to_dict
+from haystack.core.component import component
+
+from .preprocess import _preprocess_text
+
+
+@component
+class StatisticalEvaluator:
+    """
+    StatisticalEvaluator is a component that evaluates the performance of a model based on statistical metrics.
+    It's usually used in QA and Retrieval Augmented Generation (RAG) pipelines to evaluate the quality of the generated answers.
+
+    The supported metrics are:
+    - F1: Measures word overlap between predictions and labels.
+    - Exact Match: Measures the proportion of cases where prediction is identical to the expected label.
+    """
+
+    class Metric(Enum):
```

Comment:
Let's extract this and move it out the top-level namespace (`components.evaluators`). We should call it `StatisticalMetrics` (to disambiguate it from the others).